### PR TITLE
CUTEst docs

### PR DESCRIPTION
**Pull request by abelsiqueira**
Started development.

I need administration rights in CUTEst.jl to create the docs.

- [x] Use NLPModels documentation to create NLPModels api
- [ ] Document the specialized interface
- [ ] Document the core interface.

### CODE REVIEW

**Comment by dpo** [comment]
> I need administration rights in CUTEst.jl to create the docs.

Done! Many thanks!

File: docs/src/index.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+also gives a different interface for accessing the problems.
+
+## Installing
+
+Currently, this package builds its own version of CUTEst, if you have your own
+version of CUTEst, this will conflict with it. Check the issues for a fix, or
+open one for help.
+
+The following commands should automatically download NLPModels.jl and CUTEst,
+installing it.
```
Comment (dpo) [inline docs/src/index.md]: "and install them".

File: docs/src/index.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+println("gx = $( grad(nlp, nlp.meta.x0) )")
+println("Hx = $( hess(nlp, nlp.meta.x0) )")
+cutest_finalize(nlp)
+```
+
+Check the [API](api/#nlpmodels-api)
+
+There is also a specialized API which provides a more CUTEst-like interface, and
+a core API which is only a wrapper for CUTEst.
+The documentation os these is under construction.
```
Comment (dpo) [inline docs/src/index.md]: "of"

**Comment by abelsiqueira** [comment]
Basic documentation is here: [](http://juliasmoothoptimizers.github.io/CUTEst.jl/latest/)
Now for the specialized and core interfaces I'm probably gonna have to change the script and then a lot of the other files. So a different PR is better.

**Comment by dpo** [comment]
Ready to go?

**Comment by abelsiqueira** [comment]
Yes

**Comment by dpo** [comment]
Many thanks!

**Pull request merged**